src/amt3.py

```diff
@@ -469,13 +473,12 @@ def predict(self, word_indices, char_indices, task_id, train=False,
 
                 if domain_id is not None:
                     # flip the gradient when back-propagating through here
-                    adv_input = [dynet.flip_gradient(adv_in) for adv_in in concat_layer]
+                    adv_input = [dynet.flip_gradient(concat_layer[-1])] # last state
```

Comment:
We don't need to use a list here anymore as it's only one element, right? Then we also don't need to average below.